nsist/tests/test_pypi.py

```diff
@@ -8,139 +10,140 @@
-    releases = [
-        CachedRelease('astsearch-0.1.2-py3-none-any.whl'),
-        CachedRelease('astsearch-0.1.2-py3-none-win_amd64.whl'),
-    ]
-    assert_equal(wd.pick_best_wheel(releases), releases[1])
-
-    # Wrong Windows bitness
-    releases = [
-        CachedRelease('astsearch-0.1.2-py3-none-any.whl'),
-        CachedRelease('astsearch-0.1.2-py3-none-win_32.whl'),
-    ]
-    assert_equal(wd.pick_best_wheel(releases), releases[0])
-
-    # Prefer more specific Python version
-    releases = [
-        CachedRelease('astsearch-0.1.2-cp35-none-any.whl'),
-        CachedRelease('astsearch-0.1.2-py3-none-any.whl'),
-    ]
-    assert_equal(wd.pick_best_wheel(releases), releases[0])
-
-    # Prefer more specific Python version
-    releases = [
-        CachedRelease('astsearch-0.1.2-py34.py35-none-any.whl'),
-        CachedRelease('astsearch-0.1.2-py3-none-any.whl'),
-    ]
-    assert_equal(wd.pick_best_wheel(releases), releases[0])
-
-    # Incompatible Python version
-    releases = [
-        CachedRelease('astsearch-0.1.2-cp33-none-any.whl'),
-        CachedRelease('astsearch-0.1.2-py3-none-any.whl'),
-    ]
-    assert_equal(wd.pick_best_wheel(releases), releases[1])
-
-    # Prefer more specific ABI version
-    releases = [
-        CachedRelease('astsearch-0.1.2-py3-abi3-any.whl'),
-        CachedRelease('astsearch-0.1.2-py3-none-any.whl'),
-    ]
-    assert_equal(wd.pick_best_wheel(releases), releases[0])
-
-    # Incompatible ABI version
-    releases = [
-        CachedRelease('astsearch-0.1.2-cp35-abi4-win_amd64.whl'),
-        CachedRelease('astsearch-0.1.2-py3-none-any.whl'),
-    ]
-    assert_equal(wd.pick_best_wheel(releases), releases[1])
-
-    # Platform has priority over other attributes
-    releases = [
-        CachedRelease('astsearch-0.1.2-cp35-abi3-any.whl'),
-        CachedRelease('astsearch-0.1.2-py2.py3-none-win_amd64.whl'),
-    ]
-    assert_equal(wd.pick_best_wheel(releases), releases[1])
-
-def test_merge_dir_to():
-    with TemporaryDirectory() as td1, TemporaryDirectory() as td2:
-        td1 = Path(td1)
-        td2 = Path(td2)
-        with (td1 / 'ab').open('w') as f:
-            f.write(u"original")
-        with (td2 / 'ab').open('w') as f:
-            f.write(u"alternate")
-
-        (td1 / 'subdir').mkdir()
-        with (td1 / 'subdir' / 'foo').open('w'): pass
-        (td2 / 'subdir').mkdir()
-        with (td2 / 'subdir' / 'bar').open('w'): pass
-
-        merge_dir_to(td2, td1)
-
-        assert_isfile(str(td1 / 'subdir' / 'foo'))
-        assert_isfile(str(td1 / 'subdir' / 'bar'))
-        with (td1 / 'ab').open() as f:
-            assert_equal(f.read(), u"alternate")
-
-        # Test with conflicts
-        (td1 / 'conflict').mkdir()
-        with (td2 / 'conflict').open('w'): pass
-
-        with assert_raises(RuntimeError):
+# To exclude tests requiring network on an unplugged machine, use: pytest -m "not network"
+
+class TestPyPi(unittest.TestCase):
```

Comment:
Ditto here; I prefer tests as functions unless there's some shared setup/teardown (and even then, I'm increasingly using pytest fixtures).

Reply:
Pytest has a `pytest.raises` function that can be used in a similar way without the class:

https://docs.pytest.org/en/latest/assert.html#assertions-about-expected-exceptions

Reply:
As always in Python, there is a simple and elegant way to do things. Pytest reports intelligently from a condition expressed in the standard assert keyword, and for everything else there is pytest.raises.

I will update the code.